modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,403 @@
+<?php
+/**
+ * Background job class.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Class Perflab_Background_Job.
+ *
+ * Manage and run the background jobs.
+ *
+ * @since n.e.x.t
+ */
+class Perflab_Background_Job {
+	/**
+	 * Meta key for storing job name/action.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job name meta key.
+	 */
+	const JOB_NAME_META_KEY = 'perflab_job_name';
+
+	/**
+	 * Meta key for storing job data.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job data meta key.
+	 */
+	const JOB_DATA_META_KEY = 'perflab_job_data';
+
+	/**
+	 * Meta key for storing job data.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job attempts meta key.
+	 */
+	const JOB_ATTEMPTS_META_KEY = 'perflab_job_attempts';
+
+	/**
+	 * Meta key for storing job lock.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job lock meta key.
+	 */
+	const JOB_LOCK_META_KEY = 'perflab_job_lock';
+
+	/**
+	 * Meta key for storing job errors.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job errors meta key.
+	 */
+	const JOB_ERRORS_META_KEY = 'perflab_job_errors';
+
+	/**
+	 * Job status meta key.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job status meta key.
+	 */
+	const JOB_STATUS_META_KEY = 'perflab_job_status';
+
+	/**
+	 * Job status for queued jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job status queued.
+	 */
+	const JOB_STATUS_QUEUED = 'perflab_job_queued';
+
+	/**
+	 * Job status for running jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job status running.
+	 */
+	const JOB_STATUS_RUNNING = 'perflab_job_running';
+
+	/**
+	 * Job status for partially executed jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job status partial.
+	 */
+	const JOB_STATUS_PARTIAL = 'perflab_job_partial';
+
+	/**
+	 * Job status for completed jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job status commplete.
+	 */
+	const JOB_STATUS_COMPLETE = 'perflab_job_complete';
+
+	/**
+	 * Job status for failed jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string Job status failed.
+	 */
+	const JOB_STATUS_FAILED = 'perflab_job_failed';
+
+	/**
+	 * Job ID.
+	 *
+	 * @since n.e.x.t
+	 * @var int Job ID.
```

Comment:
The `@var` annotations for class properties should not have any description, per following WP docs guidelines for the most part. Especially since the extra documentation here is useless, as it's the same as above.
```suggestion
	 * @var int
```